pkg/kv/bulk/sst_batcher.go

```diff
@@ -585,18 +598,31 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 		}
 	}
 
+	currentBatchStatsCopy := b.currentBatchStats.Identity().(*backuppb.IngestionPerformanceStats)
+	currentBatchStatsCopy.Combine(&b.currentBatchStats)
+	b.currentBatchStats.Reset()
+
 	fn := func(ctx context.Context) error {
 		defer res.Release()
 		defer b.mem.Shrink(ctx, reserved)
-		if err := b.addSSTable(ctx, batchTS, start, end, data, stats, !flushAsync); err != nil {
+		if err := b.addSSTable(ctx, batchTS, start, end, data, stats, !flushAsync, currentBatchStatsCopy); err != nil {
 			return err
 		}
 		b.mu.Lock()
 		summary.DataSize += int64(size)
+		currentBatchStatsCopy.DataSize += int64(size)
 		b.mu.totalRows.Add(summary)
-		atomic.AddInt64(&b.stats.batchWaitAtomic, int64(timeutil.Since(beforeFlush)))
-		atomic.AddInt64(&b.stats.dataSizeAtomic, int64(size))
+
+		afterFlush := timeutil.Now()
+		currentBatchStatsCopy.BatchWait += afterFlush.Sub(beforeFlush)
+		currentBatchStatsCopy.Duration = afterFlush.Sub(b.lastFlush)
+		b.totalStats.Combine(currentBatchStatsCopy)
+		b.lastFlush = afterFlush
 		b.mu.Unlock()
+
+		if b.tracingSpan != nil {
```

Comment:
Does this need to be in the mutex too?

Reply:
no, because the stats copy is what's being logged.

Reply:
Right, but isn't b.tracingSpan being called from concurrent invocation of `fn`?